opal/corelib/array.rb

```diff
@@ -210,32 +210,46 @@ def +(other)
     `self.concat(other)`
   end
 
-  def -(other)
-    other = if Array === other
-              other.to_a
-            else
-              `$coerce_to(other, #{Array}, 'to_ary')`.to_a
-            end
-
+  def difference(*others)
     return [] if `self.length === 0`
-    return `self.slice()` if `other.length === 0`
+
+    others.map! do |other|
+      if Array === other
+        other.to_a
+      else
+        `$coerce_to(other, #{Array}, 'to_ary')`.to_a
+      end
+    end
+
+    result = []
+    hash = {}
 
     %x{
-      var result = [], hash = #{{}}, i, length, item;
+      var i, length, item;
+    }
 
-      for (i = 0, length = other.length; i < length; i++) {
-        $hash_put(hash, other[i], true);
+    others.each do |other|
```

Comment:
As for the `Hash` methods it's better to stay native on such important objects such as Array, can you please replace the ruby cycle with a javascript `for` loop?

Reply:
Ah, it sounds I learned in https://github.com/opal/opal/pull/2187#discussion_r58901336 😓 
I'll try!